### PR TITLE
Add jittering for Kubelet status computing

### PR DESCRIPTION
**Pull request by wojtek-t**
This PR is somewhat similar to https://github.com/kubernetes/kubernetes/pull/101652
It wasn't noticed there, as Get calls from kube-apiserver cacher are cheaper then actually write requests, but it seems these are also drifting over time and align better over time.

```release-note
NONE
```

/kind bug
/sig node
/sig scalability
/priority important-soon

@kubernetes/sig-node-bugs 

/assign @SergeyKanzhelev

### CODE REVIEW

File: pkg/kubelet/kubelet.go [hunk 1]
```diff
@@ -1469,7 +1469,10 @@ func (kl *Kubelet) Run(updates <-chan kubetypes.PodUpdate) {
 
 	if kl.kubeClient != nil {
 		// Start syncing node status immediately, this may set up things the runtime needs to run.
```
Comment (SergeyKanzhelev) [inline pkg/kubelet/kubelet.go:1471]: this comment is no longer accurate with the jitter?

File: pkg/kubelet/kubelet.go [hunk 2]
```diff
@@ -1469,7 +1469,10 @@ func (kl *Kubelet) Run(updates <-chan kubetypes.PodUpdate) {
...
-		go wait.Until(kl.syncNodeStatus, kl.nodeStatusUpdateFrequency, wait.NeverStop)
+		// Introduce some small jittering to ensure that over time the requests won't start
+		// cumulating at approximately the same time from the set of nodes due to priority and
+		// fairness effect.
+		go wait.JitterUntil(kl.syncNodeStatus, kl.nodeStatusUpdateFrequency, 0.1, true, wait.NeverStop)
```
Comment (ehashman) [inline pkg/kubelet/kubelet.go:1475]: Why are we using a bigger interval here than in #101652? (0.1 vs. 0.04)

**Review by ehashman — COMMENTED** [review]
/hold
to prevent accidental merge

File: pkg/kubelet/kubelet.go (hunk 1 above)
Comment (wojtek-t) [reply pkg/kubelet/kubelet.go]: Indeed [it wasn't before this PR too :) ]

File: pkg/kubelet/kubelet.go [hunk 3]
```diff
@@ -1469,7 +1469,10 @@ func (kl *Kubelet) Run(updates <-chan kubetypes.PodUpdate) {
 
 	if kl.kubeClient != nil {
 		// Start syncing node status immediately, this may set up things the runtime needs to run.
-		go wait.Until(kl.syncNodeStatus, kl.nodeStatusUpdateFrequency, wait.NeverStop)
+		// Introduce some small jittering to ensure that over time the requests won't start
+		// cumulating at approximately the same time from the set of nodes due to priority and
```
[the remark this answers is not in the document]
Comment (wojtek-t) [reply pkg/kubelet/kubelet.go]: done

File: pkg/kubelet/kubelet.go (hunk 2 above)
Comment (wojtek-t) [reply pkg/kubelet/kubelet.go]: I think that 0.04 is fairly small - but I don't have strong opinion here.
I switched back to 0.04 - we can always change that later...

**Review by wojtek-t — COMMENTED** [review]
@SergeyKanzhelev @ehashman - thanks for comments PTAL

File: pkg/kubelet/kubelet.go [hunk 4]
```diff
@@ -1468,8 +1468,10 @@ func (kl *Kubelet) Run(updates <-chan kubetypes.PodUpdate) {
 	go kl.volumeManager.Run(kl.sourcesReady, wait.NeverStop)
 
 	if kl.kubeClient != nil {
-		// Start syncing node status immediately, this may set up things the runtime needs to run.
-		go wait.Until(kl.syncNodeStatus, kl.nodeStatusUpdateFrequency, wait.NeverStop)
+		// Introduce some small jittering to ensure that over time the requests won't start
+		// accumulating at approximately the same time from the set of nodes due to priority and
+		// fairness effect.
+		go wait.JitterUntil(kl.syncNodeStatus, kl.nodeStatusUpdateFrequency, 0.04, true, wait.NeverStop)
```
Comment (wojtek-t) [inline pkg/kubelet/kubelet.go:1474]: To follow up on the 0.04 vs 0.1:
- withe default 10s statusUpdateFrequency, 0.1 means 1s, so we would be calculating every 9s-11s. That sounded acceptable to me.
- that said with 0.04 (so 9.6s-10.4s with default frequency) - we seem to be getting good enough spread over time for extended periods of time.
So I'm completely fine with setting it to 0.04 for consistency. If at some point we realize we need more - we can revisit it, but it doesn't seem to be the case now.

File: pkg/kubelet/kubelet.go [hunk 5]
```diff
@@ -1468,8 +1468,10 @@ func (kl *Kubelet) Run(updates <-chan kubetypes.PodUpdate) {
 	go kl.volumeManager.Run(kl.sourcesReady, wait.NeverStop)
 
 	if kl.kubeClient != nil {
-		// Start syncing node status immediately, this may set up things the runtime needs to run.
```
Comment (SergeyKanzhelev) [inline pkg/kubelet/kubelet.go:1471]: for the context on this comment: https://github.com/kubernetes/kubernetes/pull/14542#discussion_r40460183

**Review by ehashman — COMMENTED** [review]
/lgtm
will leave hold on for a bit in case node approvers want to confirm

**Review by SergeyKanzhelev — APPROVED** [review]

**Comment by wojtek-t** [comment]
Unholding based on two lgtm's above.

/hold cancel

**Pull request merged**